builtin/joined_dataset.cc

```diff
@@ -59,6 +59,30 @@ struct JoinedDataset::Itl
         //ML::compact_vector<RowHash, 2> rowHashes;   ///< Row hash from input datasets
     };
 
+     struct JoinedRowStream : public RowStream {
+
+        JoinedRowStream(JoinedDataset::Itl* source) : source(source)
+        {
+            
+        }
+
+        virtual std::shared_ptr<RowStream> clone() const{
+            auto ptr = std::make_shared<JoinedRowStream>(source);
+            return ptr;
+        }
+
+        virtual void initAt(size_t start){
+            iter = source->rows.begin() + start;
```

Comment:
Per discussion with Mathieu, on purpouse. Responsbility passed on to the caller to no go out of bounds. Will be documented in the code